src/Runner.Worker/Container/ContainerHooks/HookInput.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using System.Linq;
+
+namespace GitHub.Runner.Worker.Container.ContainerHooks
+{
+    public class HookInput
+    {
+        public HookCommand Command { get; set; }
+        public string ResponseFile { get; set; }
+        public HookArgs Args { get; set; }
+        public JToken State { get; set; }
+    }
+
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum HookCommand
+    {
+        [EnumMember(Value = "prepare_job")]
+        PrepareJob,
+        [EnumMember(Value = "cleanup_job")]
+        CleanupJob,
+        [EnumMember(Value = "run_script_step")]
+        RunScriptStep,
+        [EnumMember(Value = "run_container_step")]
+        RunContainerStep,
+    }
+    public class HookArgs { }
```

Comment:
it's a little bit weird to have an empty class...

Reply:
Refactored the class hierarchy a bit, thanks for pointing this out.